lib/rucio/common/pcache.py

```diff
@@ -1072,7 +1095,7 @@ def do_http_post(self, url, data):
             # This will retry for up to 1 hour, at 2 minute intervals
             while retry < 30:
                 try:
-                    u = urlopen(url, data=urlencode(data))
+                    u = urlopen(url, data=urlencode(data))  # type: ignore
```

Comment:
This `type: ignore` is due to the fact that the type returned by `urlencode` is not compatible with the type accepted by `data` in `urlopen`. 

I'm not sure how this actually behaves at runtime, but due to the type mismatch I'm assuming that `urlencode` shouldn't be used in this context. I think it's best to address this separately.